components/analytics-solutions-common/org.wso2.analytics.solutions.common/widgets/DateTimeRangePicker/src/widget/components/CustomTimeRangeSelector.jsx

```diff
@@ -0,0 +1,324 @@
+/* eslint-disable react/prop-types */
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from "react";
+import Moment from "moment";
+import TimePicker from "./TimePicker";
+import Button from "@material-ui/core/Button/Button";
+import Typography from "@material-ui/core/Typography/Typography";
+export default class CustomTimeRangeSelector extends React.Component {
+  state = {
+    invalidDateRange: false,
+    customRangeGranularityValue: this.props.customRangeGranularityValue,
+    startTime: this.props.startTime,
+    endTime: new Date(),
+    applyButtonBackgroundColor: "#ef6c00",
+    cancelButtonBackgroundColor: "#999"
+  };
+
+  /**
+   * Setting the start time using the custom  ranges
+   * @param{object} date
+   */
+  handleStartTimeChange = (date) => {
+    const { endTime } = this.state;
+    const startTime = date;
```

Comment:
Why assigning to another variable? You can use `date` variable instead or rename `date -> startTime`.